planner/core/fragment.go

```diff
@@ -155,6 +155,21 @@ func partitionPruning(ctx sessionctx.Context, tbl table.PartitionedTable, conds
 // single physical table means a table without partitions or a single partition in a partition table.
 func (e *mppTaskGenerator) constructMPPTasksImpl(ctx context.Context, ts *PhysicalTableScan) ([]*kv.MPPTask, error) {
 	if ts != nil {
+		// update ranges according to correlated columns in access conditions like in the Open() of TableReaderExecutor
+		hasCorCol := false
+		for _, cond := range ts.AccessCondition {
+			if len(expression.ExtractCorColumns(cond)) > 0 {
+				hasCorCol = true
+				break
+			}
+		}
+		if hasCorCol {
+			err := ts.ResolveCorrelatedColumns()
```

Comment:
we can put this logic inside line 162, to avoid an extra "hasCorCol"

Reply:
fxied